### PR TITLE
fabtests/efa: use pytest for integration test for EFA provider

### PR DESCRIPTION
**Pull request by wzamazon**
This patch is built upon https://github.com/ofiwg/libfabric/pull/7361. It introduced a set of scripts that implemented integration for EFA provider.

### CODE REVIEW

File: fabtests/pytest/efa/test_cntr.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import pytest
```
Comment (wckzhang) [inline fabtests/pytest/efa/test_cntr.py:1]: Some of these tests aren't changed at all, it feels a little bloated copying them all

File: fabtests/pytest/common.py [hunk 2]
```diff
@@ -201,3 +201,36 @@ def run(self):
         for i in range(numclient):
             check_returncode(client_process_list[i].returncode, strict)
 
+def has_cuda(ip):
```
Comment (wckzhang) [inline fabtests/pytest/common.py:204]: commit message doesn't talk about the has_cuda function

File: fabtests/pytest/efa/test_rdm.py [hunk 3]
```diff
@@ -82,22 +82,26 @@ def test_rdm_tagged_pingpong(cmdline_args, test_type, data_check_type, completio
...
-    from common import ClientServerTest
+def test_rdm_tagged_bw(cmdline_args, test_type, memory_type, completion_type):
+    from common import run_communication_test
+
     command = "fi_rdm_tagged_bw"
 
     if test_type == "short":
         command += " -I 5"
 
-    if data_check_type == "with_data_check":
```
Comment (wckzhang) [inline fabtests/pytest/efa/test_rdm.py]: Why drop the data check feature?

Comment (wckzhang) [reply fabtests/pytest/efa/test_rdm.py]: EDIT: Nvm, I see you added data check by default

**Comment by wckzhang** [comment]
It looks mostly okay (the last 3 commits, the first 4 I'll review in the other PR). How do you trigger pytests to run only efa tests for example?

File: fabtests/pytest/efa/test_cntr.py (hunk 1 above)
Comment (wzamazon) [reply fabtests/pytest/efa/test_cntr.py:1]: Yes, that is also comment from @shefty. I will try to address in new revision.

**Comment by wzamazon** [comment]
> How do you trigger pytests to run only efa tests for example?

You run `runfabtests.py -b efa server_ip client_ip`.

runfabtests.py check for `efa` directory, and only run test cases under that directory.

File: fabtests/pytest/efa/test_cntr.py (hunk 1 above)
Comment (wzamazon) [reply fabtests/pytest/efa/test_cntr.py]: In new revision, I used `from default.test_xxx import test_yyy` to import a test from default test sets. This removed duplication.

File: fabtests/pytest/common.py (hunk 2 above)
Comment (wzamazon) [reply fabtests/pytest/common.py]: This commit has been removed for future

**Comment by wzamazon** [comment]
Just updated a new revision.

One major change in this revision is that I found provider can use the following line

```
from default.test_xxx import test_yyy
```
to import a test from default test sets, instead of copying the source code. In this case, any change to the test cases `test_yyy` in `default/test_xxx.py` will be picked up by the provider.

This change reduced code duplication by not copying code from `default`.

**Review by wckzhang — APPROVED** [review]

**Pull request merged**